### ISSUE TITLE
Reducing the dependency footprint

### ISSUE THREAD

**Issue by danielwhite** [flashmob/go-guerrilla#94]
Currently I've been building a prototype to use this to provide an SMTP adapter to an internal system and I'm really loving the changes to provide this as a library.

The problem I'm facing with gaining more acceptance are the dependencies that are unrelated to the primary problem.

Is there is any appetite for contributions to reduce this footprint?

Specifically:

- MySQL: provide a compatible interface or use a separate repository (like `maildir-processor`)
- Redis: provide a compatible interface or use a separate repository (like `maildir-processor`)
- logrus: provide a compatible interface to logrus, but only directly reference in `cmd/`

From what I can tell, [`cmd/guerrillad`](https://github.com/flashmob/go-guerrilla/tree/master/cmd/guerrillad) is excluded from dependency resolution (at least by [`golang/dep`](https://github.com/golang/dep)), so it definitely can continue to import the above packages to provide a fully featured default server.

**Comment by flashmob** [comment]
yeah, life got in the way a bit. Very close though, it would be interesting
to see how it performs. Perhaps one day!
